### ISSUE TITLE
[BUG] Betterjoy has broken my integrated bluetooth

### ISSUE THREAD

**Issue by weishen1121** [Davidobot/BetterJoy#970]
I got everything setup with joycons, was playing a game on Ryujinx with both joycons working, then I went to close Ryujinx and my computer froze. I waited a few minutes but it was still frozen and not responding to anything. Restarted my computer and the bluetooth was constantly connecting/disconnecting. Only fix is to disable my bluetooth with Device Manager and use a different USB bluetooth adapter. 

I've been troubleshooting for 2 hours with no success. I've got an X570 AORUS ELITE WIFI motherboard. I've tried uninstalling and deleting drivers through Device Manager for both bluetooth and wireless, disabling internet, rebooting and re-installing drivers from manufacture site. I've tried unpairing all bluetooth devices with Bluetooth Command Line Tools. I've tried deleting all bluetooth devices from regedit. No matter what I do, the integrated bluetooth keeps connecting/disconnecting until it is manually Disabled under Device Manager. 

Googling leads me to believe the only fix is to re-install windows... does anyone else have any ideas? 
I really don't understand why an app freezing my computer, leading to a hard reboot could ever cause such an issue but my brain is about to melt. Please.. any help?

**Comment by weishen1121** [comment]
I came across that while googling but I don't actually have HidGuardian installed.

Do you have any links to any guides that go over a full bluetooth re-install or could it be during the freeze/force restart something fried the integrated bluetooth hardware part of my motherboard? I'm thinking it's fried because it keeps saying something along the lines of 'Device xxxx had a problem starting' under event viewer unless the Intel Wireless Bluetooth entry under Device Manager is disabled.

It really seems like the only option is to buy a separate PCI-E bluetooth card or bluetooth dongle and I'm worried this might happen all over again to the new card/dongle.

**Comment by weishen1121** [comment]
> The only thing BetterJoy does that vaguely could lead to this problem is use HidGuardian. There are tutorials online on how to fully remove this.

Although I did find HidHide from DS4windows but uninstalling that did not help :(

**Comment by weishen1121** [comment]
Great update - weird solution but computer just needed to be shut off and unplugged for ~8 hours! 

I had read yesterday that shutting down the computer and unplugging the power worked for some, so I tried it yesterday but only shutting it down/unplugging from power for a few minutes at most. I had read a post where someone said 24 hours but I was very doubtful. I still went ahead and shut it down last night and unplugged it while I slept. It's been about 8 hours. When I turned it on this morning, my Intel Wireless Bluetooth entry under Device Manager was still disabled, so I uninstalled + deleted software in one last attempt. Scanned for hardware changes and bam, it actually worked without disconnecting/connecting again! I had also removed all my paired bluetooth devices and they were gone, but as soon as I scanned for hardware changes, they all came back and all listed under the same bluetooth section. (the bluetooth section only had the Intel Wireless Bluetooth entry under Device Manager previously) Just to be sure, I removed all my bluetooth pairings and restarted the computer one last time. Paired my DS from PS5 and everything seems to be working! Was even able to turn on Hide DS4 Controller option under DS4windows with no issues! (Although I'm pretty sure this is HidHide from DS4windows, not HidGuardian)

I'm not exactly sure how this works out, but hopefully this will help someone else in this position as well.
Just leave it alone, unplugged from the power for ~8 hours and hopefully won't need to go insane troubleshooting for 4+ hours lmao.
